### PR TITLE
Add a retry mechanism configurable by a retryCount property

### PR DESCRIPTION
**Pull request by fcduarte**
@kattrali Please review it. Could you also generate a SNAPSHOT version at Bintray? Some of my team members would be able to test it and then we'll see if we're able to get rid of the `MultiEntityBuilder` error.

Thanks!

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -18,13 +18,15 @@ compileGroovy {
...
 
+def ApacheHttpComponentsVersion = '4.5.2'
+
 // Build dependencies
 dependencies {
     compile gradleApi()
     compile localGroovy()
-    compile 'com.android.tools.build:gradle:1.5.0'
-    compile 'org.apache.httpcomponents:httpclient:4.3.6'
-    compile 'org.apache.httpcomponents:httpmime:4.3.6'
```
Comment (kattrali) [inline build.gradle]: Per the previous discussion about the dependency versioning, I’d rather introduce new versions of httpclient separately from this feature, as it breaks the configuration of existing users. The plugin also supports gradle 1.5.0+, so I don’t see a reason to change the lower bound.

**Comment by kattrali** [comment]
Can you separate the `retryCount` mechanism from the dependency upgrading so we can bring these in separately? The retry feature can be a small addition in a new minor release without changing the dependencies.

**Comment by kattrali** [comment]
Got this sorted! Making a snapshot shortly.

**Comment by kattrali** [comment]
Actually, cancel that - JCenter does not accept snapshot builds. I can push one to maven central if that helps, otherwise I’m drafting the 2.3.0 release in #28.

**Comment by fcduarte** [comment]
Whichever works better for you, I just wanted to help on the test process to avoid any bugs like we had on #25

**Comment by kattrali** [comment]
Ah, no worries then. I tried environment isolation with 4.3.6 (and ran `gradle clean` a ridiculous number of times), and it works.

**Pull request closed without merging**